### PR TITLE
Change the way we figure out container to read the artifacts from

### PR DESCRIPTION
**Pull request by viveksinghggits**
## Change Overview

While running the kanister function, to read the artifact that the phase has produced we get the logs of the kanister function pod (let's say `KubeTask`). If there is just one container in the KubeTask pod in that case things would work but if there are multiple contianers in the pod (let's say service mesh injected one), in that case the logic that we have to figure out the container name was not efficient. If was relying on the first container of the pod.

This PR changes that logic to read the container name from `podOptions` or fallback to the default container name. This should work in all the cases, if container name is not specified while creating `podOptions` we will use default container name everywhere otherwise we will use the container name that is specified in the podOptions.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [x] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by viveksinghggits** [comment]
@kale-amruta FYI, I have added some code related to repo server as well.

**Comment by viveksinghggits** [comment]
@pavannd1 the main change is in `pod_controller.go` that needs to be reviewed. Rest of the change is just because of that.

File: pkg/kube/pod_controller.go [hunk 1]
```diff
@@ -211,26 +211,20 @@ func (p *podController) StopPod(ctx context.Context, stopTimeout time.Duration,
...
-// getContainerName returns container name, which should be passed to
-// operations that require it.
-// If the container name was specified in podOptions, it will be used.
-// Otherwise, the first container name from specs will be taken as best effort
-// (when pods are created with sidecars, sidecar containers are placed after
-// main container).
-func (p *podController) getContainerName() string {
-	if p.podOptions.ContainerName != "" {
-		return p.podOptions.ContainerName
+func ContainerNameFromPodOpts(po *PodOptions) string {
```
Comment (pavannd1) [inline pkg/kube/pod_controller.go:214]: If this is not limited to `podController`, we should move it to `pod.go`.
Additionally, I preferred the previous name. It was generic. It's getting the name from `PodOptions` only when available. So, `GetContainerName` or `ContainerNameFromPodOptsOrDefault` would be more appropriate.
The comment section should be modified and added back as well.

File: pkg/kube/pod_controller_test.go [hunk 2]
```diff
@@ -309,3 +309,27 @@ func (s *PodControllerTestSuite) TestPodControllerGetCommandExecutorAndFileWrite
 		tc(pcp, pc)
 	}
 }
+
+func (s *PodControllerTestSuite) TestContainerNameFromPodOpts(c *C) {
```
Comment (pavannd1) [inline pkg/kube/pod_controller_test.go:313]: If we move the helper in `pod.go`, this should be moved as well

**Review by pavannd1 — APPROVED** [review]

**Pull request merged**